### PR TITLE
docs: guide for dependency imports

### PR DESCRIPTION
**Pull request by BenjaminMummery**
[ORQSDK-487]

# The problem

Imports to tasks can be complicated. The addition of PythonImports may make this more confusing for our users.

# This PR's solution

A user facing docs page detailing the best practices for workflow dependencies.

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/internal/docs>[!]:`

[ORQSDK-487]: https://zapatacomputing.atlassian.net/browse/ORQSDK-487?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/guides/dependency-installation.rst [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+For workflows running remotely the ``GitHubImport`` importer should be used unless there is a specific reason to do otherwise (i.e. the code is hosted somewhere other than GitHub). ``GitImport.infer()`` is highly user-specific and so should never be used in a collaborative context.
+
+For workflows running locally, the task decorator can typically be used without specifying the source importer as the SDK will set this appropriately for the context:
+
+.. code-block:: python
+    @sdk.task
+    def my_task():
+        pass
+
+will perform a local import unless running in a Jupyter Notebook (or other interactive session) in which case am inline import will be performed.
```
Comment (jamesclark-Zapata) [inline docs/guides/dependency-installation.rst:223]: **thought:** ‏Should this be first?

**Review by jamesclark-Zapata — CHANGES_REQUESTED** [review]
woohoo! Thanks for tackling these docs. This will certainly help our users.

A few minor comments + I'm not sure on the word "ingests" in some places.

Can you ask Tim or some internal users (from the recent Slack thread) to check this too?

File: docs/guides/dependency-installation.rst [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+    * - Importing From
+      - Importer
+      - Runtime
+      - Notes
+    * - GitHub repo
+      - ``GitHubImport``
+      - remote
+      - On Ray runtimes, secrets can be used to access private repos.
+    * - Non-github git repo without a local clone
```
[the remark this answers is not in the document]
Comment (BenjaminMummery) [reply docs/guides/dependency-installation.rst]: This made more sense when the table included `.infer`, I should have removed it. Fixed in
[f419250](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/74/commits/f41925080adb712979d7961a0c5eff4822c9aef6)

File: docs/guides/dependency-installation.rst [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+      - ``GitImport``
+      - remote
+      -
+    * - PyPi
+      - ``PythonImports``
+      - remote
+      -
+    * - Jupyter Notebook
+      - ``InlineImport``
+      - local
```
[the remark this answers is not in the document]
Comment (BenjaminMummery) [reply docs/guides/dependency-installation.rst]: Fixed in [d594f39](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/74/commits/d594f39c00cbbd912d31fa490713bdc0c5339fb6)

File: docs/guides/dependency-installation.rst (hunk 1 above)
Comment (BenjaminMummery) [reply docs/guides/dependency-installation.rst]: I wAlmost certainly, especially with the quick reference tl;dr up top. Fixed in [2337332](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/74/commits/2337332f42a2571eead93827cf1d58ae61ec6a0e)

File: docs/guides/dependency-installation.rst [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+      - ``GitHubImport``
+      - remote
+      - On Ray runtimes, secrets can be used to access private repos.
+    * - Non-github git repo without a local clone
+      - ``GitImport``
+      - remote
+      -
+    * - PyPi
+      - ``PythonImports``
+      - remote
```
[the remark this answers is not in the document]
Comment (BenjaminMummery) [reply docs/guides/dependency-installation.rst]: Updated in [4502ba6](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/74/commits/4502ba653b15e3828263d6f7c35068b9eeaefffc) so this is now correct w.r.t. the current behaviour.

File: docs/guides/dependency-installation.rst [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+            "zapatacomputing/my_source_repo",
+            git_ref = "feat/my-feature-branch",
+            username = "my-github-username",
+            personal_access_token = sdk.Secret("my_pat")
+        )
+    )
+    def demo_task():
+        pass
+
+The PAT is ingested from the portal secrets management facility at run time, and installation of the import proceeds as usual.
```
[the remark this answers is not in the document]
Comment (BenjaminMummery) [reply docs/guides/dependency-installation.rst]: [2b0e649](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/74/commits/2b0e6498e022a5ff3b92ab831d69af6fe78db3c1)

**Review by jamesclark-Zapata — APPROVED** [review]

**Pull request merged**